src/generators/android/android.ts

```diff
@@ -0,0 +1,328 @@
+	},
+	generateArray: async (client, schema, items, parentPath) => {
+		return {
+			...defaultPropertyContext(client, schema, `List<${items.type}>`, parentPath),
+		}
+	},
+	generateObject: async (client, schema, properties, parentPath) => {
+		const property = defaultPropertyContext(client, schema, JavaType.Object, parentPath)
+		let object: AndroidObjectContext | undefined
+
+		if (properties.length > 0) {
+			const className = client.namer.register(schema.name, 'class', {
+				transform: (name: string) => {
+					return upperFirst(camelCase(name))
+				},
+			})
+
+			property.type = className
+			object = {
+				name: className,
+			}
+		}
+
+		return { property, object }
+	},
+	generateUnion: async (client, schema, _, parentPath) => {
+		// TODO: support unions
+		return defaultPropertyContext(client, schema, 'Object', parentPath)
+	},
+	generateTrackCall: async (client, schema) => {
+		const { properties } = getPropertiesSchema(schema)
+		return {
+			functionName: client.namer.register(schema.name, 'function->track', {
+				transform: camelCase,
+			}),
+			propsParam: !!properties.length,
+		}
+	},
+	generateRoot: async (client, context) => {
+		await Promise.all([
+			client.generateFile(
+				'TypewriterAnalytics.java',
+				'generators/android/templates/analytics.java.hbs',
+				context
+			),
+			client.generateFile(
+				'TypewriterUtils.java',
+				'generators/android/templates/typewriterUtils.java.hbs',
+				context
+			),
+			client.generateFile(
+				'SerializableProperties.java',
+				'generators/android/templates/abstractSerializableClass.java.hbs',
+				context
+			),
+			...context.objects.map(o =>
+				client.generateFile(`${o.name}.java`, 'generators/android/templates/class.java.hbs', o)
+			),
+		])
+	},
+}
+
+interface Param {
+	hasParam: boolean
+	name: string
+	type: string
+}
+
+interface Arg {
+	inUse: boolean
+	execution: string
+	fallback?: string
+}
+
+enum Modifier {
+	FinalNullable = 'final @Nullable',
+	FinalNonNullable = 'final @NonNull',
+}
+
+enum Separator {
+	Comma = ', ',
+	Indent = '  ',
+	NewLineIndent = '      ',
+}
+
+enum Properties {
+	ToProperties = 'props.toProperties()',
+	Create = 'new Properties()',
+}
+
+enum Options {
+	MergeOptions = 'TypewriterUtils.addTypewriterContext(options)',
+	Create = 'TypewriterUtils.addTypewriterContext()',
+}
+
+function defaultPropertyContext(
+	client: GeneratorClient,
+	schema: Schema,
+	type: string,
+	namespace: string
+): AndroidPropertyContext {
+	return {
+		name: client.namer.register(schema.name, namespace, {
+			transform: camelCase,
+		}),
+		type,
+		modifiers:
+			!schema.isRequired || !!schema.isNullable
+				? Modifier.FinalNullable
+				: Modifier.FinalNonNullable,
+		isVariableNullable: !schema.isRequired || !!schema.isNullable,
+		nonNullPayload: !schema.isNullable,
+	}
+}
+
+function generateBuilderFunctionSignature(name: string, modifiers: string, type: string): string {
+	return `public Builder ${name}(${modifiers} ${type} ${name})`
```

Comment:
Could this just live in the template itself? Personally, I prefer to templatize as much as possible, and only divert to a handlerbars function when absolutely necessary because it adds indirection :)